// react-richtext/lib/ui/LinkImagePopover.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _editorMethods = require('../functions/editorMethods');

var _Button = require('./Button');

var _Button2 = _interopRequireDefault(_Button);

var _styles = require('../assets/styles');

var _styles2 = _interopRequireDefault(_styles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

//import { MASTER_EDITOR } from '../lib/EditorToolbarConfig'; //not currently needed as this is not controlled by config


var LinkImagePopover = function (_Component) {
    _inherits(LinkImagePopover, _Component);

    function LinkImagePopover(props) {
        _classCallCheck(this, LinkImagePopover);

        var _this = _possibleConstructorReturn(this, Object.getPrototypeOf(LinkImagePopover).call(this, props));

        _this.state = {
            showPopover: false,
            popoverBasis: {},
            inputRef: ''
        };

        _this.toggleLink = _editorMethods.toggleLink.bind(_this);
        _this.addImageLink = _editorMethods.addImageLink.bind(_this);
        _this.updateLinkInputValue = _editorMethods.updateLinkInputValue.bind(_this);
        _this.togglePopover = _editorMethods.togglePopover.bind(_this);

        _this.closePopoverOnResize = _editorMethods.closePopoverOnResize.bind(_this);
        _this.handleOnKeyPress = _editorMethods.handleOnKeyPress.bind(_this, _this.addImageLink);
        return _this;
    }

    _createClass(LinkImagePopover, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            window.addEventListener('resize', this.closePopoverOnResize);
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            window.removeEventListener('resize', this.closePopoverOnResize);
        }
    }, {
        key: 'renderPopover',
        value: function renderPopover() {
            var popoverLinkStyles = this.props.popoverLinkStyles;
            var popoverBasis = this.state.popoverBasis;


            return _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(
                    'form',
                    {
                        style: Object.assign({}, popoverBasis, popoverLinkStyles.popoverContainer, popoverLinkStyles.basePopoverContainer)
                    },
                    _react2.default.createElement(
                        'div',
                        { style: popoverLinkStyles.inner },
                        _react2.default.createElement('input', {
                            type: 'text',
                            placeholder: 'https://example.com/',
                            style: popoverLinkStyles.input,
                            value: this.state.inputRef,
                            onChange: this.updateLinkInputValue,
                            onKeyPress: this.handleOnKeyPress
                        }),
                        _react2.default.createElement(
                            'span',
                            { style: popoverLinkStyles.buttonGroup },
                            _react2.default.createElement(_Button2.default, {
                                label: 'Cancel',
                                iconName: 'cancel',
                                onClick: this.toggleLink,
                                passedButtonStyles: popoverLinkStyles.formButtons
                            }),
                            _react2.default.createElement(_Button2.default, {
                                label: 'Submit',
                                iconName: 'accept',
                                type: 'submit',
                                onClick: this.addImageLink,
                                passedButtonStyles: popoverLinkStyles.formButtons
                            })
                        )
                    )
                ),
                _react2.default.createElement('div', { style: popoverLinkStyles.basePopoverBackdrop, onClick: this.toggleLink })
            );
        }
    }, {
        key: 'render',
        value: function render() {
            var _props = this.props;
            var label = _props.label;
            var popoverLinkStyles = _props.popoverLinkStyles;

            var renderPopover = this.state.showPopover ? this.renderPopover() : null;

            return _react2.default.createElement(
                'div',
                { style: popoverLinkStyles.baseContainer },
                _react2.default.createElement(
                    'div',
                    { style: popoverLinkStyles.basePopoverTrigger },
                    _react2.default.createElement(
                        _Button2.default,
                        {
                            label: label,
                            onClick: this.toggleLink
                        },
                        renderPopover
                    )
                )
            );
        }
    }]);

    return LinkImagePopover;
}(_react.Component);

LinkImagePopover.propTypes = {
    popoverLinkStyles: _react.PropTypes.object,
    editorState: _react.PropTypes.object,
    label: _react.PropTypes.string
};

LinkImagePopover.defaultProps = {
    popoverLinkStyles: Object.assign({}, _styles2.default.popoverLinkStyles, _styles2.default.baseStyles)
};

exports.default = LinkImagePopover;
