// react-richtext/lib/ui/LocalImageUpload.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Button = require('./Button');

var _Button2 = _interopRequireDefault(_Button);

var _editorMethods = require('../functions/editorMethods');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var LocalImageUpload = function (_Component) {
    _inherits(LocalImageUpload, _Component);

    function LocalImageUpload(props) {
        var _Object$getPrototypeO;

        _classCallCheck(this, LocalImageUpload);

        var _this = _possibleConstructorReturn(this, (_Object$getPrototypeO = Object.getPrototypeOf(LocalImageUpload)).call.apply(_Object$getPrototypeO, [this, props].concat(Array.prototype.slice.call(arguments))));

        _this.state = {
            fileInput: ''
        }, _this.getCurrentBlockType = _editorMethods.getCurrentBlockType.bind(_this);
        _this.uploadImage = _editorMethods.uploadImage.bind(_this);
        _this.fileInput = _editorMethods.fileInput.bind(_this);
        _this.insertImage = _editorMethods.insertImage.bind(_this);
        return _this;
    }

    _createClass(LocalImageUpload, [{
        key: 'render',
        value: function render() {
            var _props = this.props;
            var label = _props.label;
            var editorState = _props.editorState;

            var blockType = this.getCurrentBlockType();

            return _react2.default.createElement(
                'span',
                null,
                _react2.default.createElement('input', { type: 'file', ref: 'fileInput', style: { display: 'none' },
                    onChange: this.fileInput }),
                _react2.default.createElement(_Button2.default, {
                    label: label,
                    selected: blockType,
                    onClick: this.uploadImage
                })
            );
        }
    }]);

    return LocalImageUpload;
}(_react.Component);

LocalImageUpload.propTypes = {
    editorState: _react.PropTypes.object,
    label: _react.PropTypes.string
};

exports.default = LocalImageUpload;
