// react-richtext/lib/lib/EditorToolbar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _draftJs = require('draft-js');

var _EditorToolbarConfig = require('./EditorToolbarConfig');

var _LinkImagePopover = require('../ui/LinkImagePopover');

var _LinkImagePopover2 = _interopRequireDefault(_LinkImagePopover);

var _LinkInputPopover = require('../ui/LinkInputPopover');

var _LinkInputPopover2 = _interopRequireDefault(_LinkInputPopover);

var _ToolbarButtons = require('../ui/ToolbarButtons');

var _ToolbarButtons2 = _interopRequireDefault(_ToolbarButtons);

var _ColorsButtonPopover = require('../ui/ColorsButtonPopover');

var _ColorsButtonPopover2 = _interopRequireDefault(_ColorsButtonPopover);

var _BasicDropdowns = require('../ui/BasicDropdowns');

var _BasicDropdowns2 = _interopRequireDefault(_BasicDropdowns);

var _UndoRedo = require('../ui/UndoRedo');

var _UndoRedo2 = _interopRequireDefault(_UndoRedo);

var _LocalImageUpload = require('../ui/LocalImageUpload');

var _LocalImageUpload2 = _interopRequireDefault(_LocalImageUpload);

var _styles = require('../assets/styles');

var _styles2 = _interopRequireDefault(_styles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var EditorToolbar = function (_Component) {
    _inherits(EditorToolbar, _Component);

    function EditorToolbar(props) {
        _classCallCheck(this, EditorToolbar);

        return _possibleConstructorReturn(this, Object.getPrototypeOf(EditorToolbar).call(this, props));
    }

    _createClass(EditorToolbar, [{
        key: 'render',
        value: function render() {
            var editorToolbarStyles = this.props.editorToolbarStyles;

            return _react2.default.createElement(
                'div',
                { style: editorToolbarStyles.toolbarContainer },
                _react2.default.createElement(_BasicDropdowns2.default, {
                    basicDropdownSrc: _EditorToolbarConfig.MASTER_EDITOR.basicDropdowns,
                    editorState: this.props.editorState,
                    onChange: this.props.onChange
                }),
                _react2.default.createElement(_ColorsButtonPopover2.default, {
                    popoverColorsSrc: _EditorToolbarConfig.MASTER_EDITOR.popoverColors,
                    editorState: this.props.editorState,
                    onChange: this.props.onChange
                }),
                _react2.default.createElement(_ToolbarButtons2.default, {
                    buttonsSrc: _EditorToolbarConfig.MASTER_EDITOR.basicButtons,
                    editorState: this.props.editorState,
                    onChange: this.props.onChange
                }),
                _react2.default.createElement(_LinkInputPopover2.default, {
                    label: "Link",
                    editorState: this.props.editorState,
                    onChange: this.props.onChange
                }),
                _react2.default.createElement(_UndoRedo2.default, {
                    label: "Undo",
                    editorState: this.props.editorState,
                    onChange: this.props.onChange
                }),
                _react2.default.createElement(
                    'div',
                    { style: _styles2.default.buttongroup },
                    _react2.default.createElement(_LinkImagePopover2.default, {
                        label: "Image",
                        editorState: this.props.editorState,
                        onChange: this.props.onChange
                    }),
                    _react2.default.createElement(_LocalImageUpload2.default, {
                        label: "LocalImage",
                        editorState: this.props.editorState,
                        onChange: this.props.onChange
                    })
                )
            );
        }
    }]);

    return EditorToolbar;
}(_react.Component);

exports.default = EditorToolbar;


EditorToolbar.propTypes = {
    editorToolbarStyles: _react.PropTypes.object,
    editorState: _react.PropTypes.object,
    onChange: _react.PropTypes.func
};
EditorToolbar.defaultProps = {
    editorToolbarStyles: _styles2.default.editorToolbarStyles
};
